File: src/auction/auction-settlement-service/src/main/java/com/enterprise/auctionsettlement/Application.java
package com.enterprise.auctionsettlement;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@SpringBootApplication
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @RestController
    public static class HealthController {
        @GetMapping("/healthz")
        public Map<String, String> healthz() {
            return Map.of("status", "ok");
        }
    }
}

File: src/rental/lease-service/src/main/kotlin/com/enterprise/lease/Application.kt
package com.enterprise.lease

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}

@RestController
class HealthController {
    @GetMapping("/healthz")
    fun healthz(): Map<String, String> = mapOf("status" to "ok")
}
